SHARE-477 PUT /notifications/read

Kropiunig opened the pull request:
-implemented notification/read API
-removed old API
-replaced usages of old API with new API
-wrote test for API

---
### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroweb-Symfony/blob/develop/.github/contributing.md) and [wiki pages](https://github.com/Catrobat/catroweb-Symfony/wiki/) of this repository.

- [x] Include the name and id of the Jira ticket in the PR’s title eg.: `SHARE-666 The devils ticket`
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no warnings and errors 
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Verify that all tests are passing (CI), if not please state the test cases in the [section](#Tests) below
- [x] Perform a self-review of the changes
- [x] Stick to the project’s git workflow (rebase and squash your commits)
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] Put your ticket into the `Code Review` section in [Jira](https://jira.catrob.at/)
- [x] Post a message in the *#catroweb* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer
- [x] Check that your pull request has been successfully deployed to https://web-test-1.catrob.at/

### Additional Description
`TODO: Add additional information that is not in your commit-message here`

### Tests - additional information
`TODO: add additional information about testruns here`

Kropiunig committed:
SHARE-477 PUT /notifications/read

-implemented notification/read API
-removed old API
-replaced usages of old API with new API
-wrote test for API

dmetzner commented on src/Api/Services/Notifications/NotificationsApiProcessor.php:
```diff
@@ -3,7 +3,41 @@
+   */
+  public function markAllAsSeen(User $user): void
+  {
+    $notifications = $this->notification_repository->findBy(['user' => $user]);
```
I know it already was coded that way, however, why would we need to fetch all notifications one by one to check their status and change it. A simple update query for all unseen notifications of the user would be enough, no need for any PHP logic. At least one should query only the unseen notifications of the user...


if you got some time I would highly appreciate if you could fix this performance issue. :)

Kropiunig committed:
SHARE-477 PUT /notifications/read

-implemented notification/read API
-removed old API
-replaced usages of old API with new API
-wrote test for API

dmetzner commented on src/DB/EntityRepository/User/Notification/NotificationRepository.php:
```diff
@@ -76,4 +76,30 @@ public function getFollowNotificationForUser(User $owner = null, User $follow_fr
+    $qb = $this->_em->createQueryBuilder();
+
+    $qb
+      ->select('n')
```
update  bsp: https://stackoverflow.com/questions/4337751/doctrine-2-update-query-with-query-builder
